presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/RetryingConnectionFactory.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.jdbc;
+
+import com.google.common.base.Throwables;
+import io.prestosql.spi.PrestoException;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.FailsafeException;
+import net.jodah.failsafe.RetryPolicy;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.SQLRecoverableException;
+
+import static java.time.temporal.ChronoUnit.MILLIS;
+import static java.time.temporal.ChronoUnit.SECONDS;
+import static java.util.Objects.requireNonNull;
+
+public class RetryingConnectionFactory
+        implements ConnectionFactory
+{
+    private static final RetryPolicy<Object> RETRY_POLICY = new RetryPolicy<>()
+            .withMaxDuration(java.time.Duration.of(30, SECONDS))
+            .withMaxAttempts(10)
```

Comment:
10 is overkill. Make it 3.

Reply:
I wanted to last it longer. Now it will retry at:
50, 150, 250, 450, 850, 1650, 3250, 8250, 13250, 18250 (notice that there is not included time that is required do the connection open, typically it is like from 10ms to 100ms).

For example if there is some kind of outage waiting 18 seconds could help, while 250 millis might be too short. Anyway, it is heuristic. That way I guess is safer but without causing big user experience impact. 

WDYT?

Reply:
It probably do not matter much. I would prefer to have less of those. First one immediately after failure. And next one after >=1s. If that is not easily achievable you can leave it as is, yet I am a bit afraid that having too many of those, too frequently can overwhelm database server.

Reply:
Ok. I made it less frequent and there are less calls. From 10 to 5, but still it has the same time range up to 30 seconds.

Reply:
thx